agent/api/task.go

```diff
@@ -220,13 +220,28 @@ func (task *Task) dockerConfig(container *Container) (*docker.Config, *DockerCli
 		Env:          dockerEnv,
 		Memory:       dockerMem,
 		CPUShares:    task.dockerCpuShares(container.Cpu),
-		Labels: map[string]string{
-			"com.amazonaws.ecs.task-arn":                task.Arn,
-			"com.amazonaws.ecs.container-name":          container.Name,
-			"com.amazonaws.ecs.task-definition-family":  task.Family,
-			"com.amazonaws.ecs.task-definition-version": task.Version,
-		},
+		Labels:       map[string]string{},
 	}
+
+	if container.DockerConfig.Config != nil {
+		err := json.Unmarshal([]byte(*container.DockerConfig.Config), &config)
+		if err != nil {
+			return nil, &DockerClientConfigError{"Unable decode given docker config: " + err.Error()}
+		}
+	}
+
+	// Augment labels with some metadata from the agent. Explicitly do this last
+	// such that it will always override duplicates in the provided raw config
+	// data.
+	for key, value := range map[string]string{
```

Comment:
Can you just change this to

```
config.Labels["com.amazonaws.ecs.task-arn"] = task.Arn
config.Labels["com.amazonaws.ecs.container-name"] = container.Name
config.Labels["com.amazonaws.ecs.task-definition-family"] = task.Family
config.Labels["com.amazonaws.ecs.task-definition-version"] = task.Version
```
?

I don't think the loop is particularly readable or adds significant value.